### PR TITLE
Fix crash when Clementine lists albums of Ampache et remove subsonic songs when the user disconnects

### PR DESCRIPTION
**Pull request by Chocobozzz**
I think it's better if the songs  are removed when the user changes the subsonic server or disconnects

### CODE REVIEW

File: src/internet/subsonic/subsonicservice.cpp [hunk 1]
```diff
@@ -411,13 +416,15 @@ void SubsonicLibraryScanner::OnGetAlbumListFinished(QNetworkReply* reply,
...
 
   int albums_added = 0;
-  reader.readNextStartElement();
-  Q_ASSERT(reader.name() == "albumList2");
-  while (reader.readNextStartElement()) {
-    Q_ASSERT(reader.name() == "album");
-    album_queue_ << reader.attributes().value("id").toString();
-    albums_added++;
-    reader.skipCurrentElement();
+  if(skip_read_albums == false) {
```
Comment (hatstand) [inline src/internet/subsonic/subsonicservice.cpp]: `if (!skip_read_albums)`

File: src/internet/subsonic/subsonicservice.cpp [hunk 2]
```diff
@@ -411,13 +416,15 @@ void SubsonicLibraryScanner::OnGetAlbumListFinished(QNetworkReply* reply,
...
+    reader.readNextStartElement();
+    Q_ASSERT(reader.name() == "albumList2");
```
Comment (hatstand) [inline src/internet/subsonic/subsonicservice.cpp]: You should really handle invalid input better than this.

**Comment by Chocobozzz** [comment]
Do you think we should handle all potential invalid tags ? I didn't touch all other assertions.

File: src/internet/subsonic/subsonicservice.cpp [hunk 3]
```diff
@@ -411,13 +416,19 @@ void SubsonicLibraryScanner::OnGetAlbumListFinished(QNetworkReply* reply,
...
-    reader.skipCurrentElement();
+  if(!skip_read_albums) {
+    reader.readNextStartElement();
+    if(reader.name() != "albumList2") {
+      qLog(Warning) << "Subsonic parsing error: albumList2 tag expected. Aborting scan.";
+      return;
+    }
+
+    while (reader.readNextStartElement()) {
+      Q_ASSERT(reader.name() == "album");
```
Comment (hatstand) [inline src/internet/subsonic/subsonicservice.cpp]: You shouldn't assert on invalid input, just recognise that it's invalid and give up parsing.

**Comment by Chocobozzz** [comment]
So I replaced all the assertions of the file.

**Comment by hatstand** [comment]
Please run `make format`

**Comment by Chocobozzz** [comment]
Sorry, I forgot it.

**Pull request merged**